_posts/2020-11-30-developer-survey-2020.md

```diff
@@ -0,0 +1,99 @@
+
+Kotlin was the most wanted language here, with 29% of the respondents saying they would like to use Kotlin in the future. As such, people using Kotlin currently seem to want to continue using it and others would like to start using it. As such, Kotlin seems like a good choice for future projects.
+
+Clojure came third, with 22% wanting to work with it in future. TypeScript came third with 18%. With TypeScript usage having increased a lot in two years, the interest is still growing.
+
+For languages not currently in wide use, Rust was a clear winner, with 8% of respondents wanting to use it. With appropriate projects, we could very well see Rust move to the used languages category for the next survey.
+
+## Frontend libraries and frameworks
+![Frontend frameworks](/img/developer-survey-2020/frontend_frameworks_and_libraries.png)
+
+React rules the frontend at the moment. 66% who answered to the question about used frontend libraries or frameworks reported having used React in the last year. All the competion (Angular, Aurelia, Vue) combined actually only just barely reached React's numbers. Over half of the React users also used Redux with it and a half also used React Router. ClojureScript developers rely on Reagent as their React library and then most of them also use re-frame for state-management.
+
+For ready-made styles or React components, about a fifth of the answerers reported using Bootstrap or Material-UI.
+
+## Backend libraries, frameworks and tools
+
+For backend libraries frameworks and tools the amount of things that could be included here made it a hard question to answer. Interpreting the results, the most used frameworks and libraries were:
+
+1. Spring Boot
+2. Spring Framework
+3. Log4J
+
+The most used tool was Docker, which in the context of this question means that it is used at least for setting up development environments. Integration frameworks in use were Camel, Mule ESB and Dell Boomi.
+
+Other common answers were Node.js, Hibernate, .NET Core, ASP.NET, and parts of Clojure web stack such as Compojure, Reitit and Ring.
+
+## Platforms
+
+Where do we deploy our software? 50% reported that their software runs on Linux. That Linux can of course then be running in traditional servers or in some kind of cloud service. The most used cloud platform was AWS, followed by Azure which had a bit over half the users as AWS. 23% of respondents reported also using a serverless computing platform of their cloud service provider.
+
+When asked what platforms we would like to use in the future, the top 3 consisted of the three big cloud services. The most wanted was AWS, followed by Azure and then Google Cloud Platform. This mirrored the current usage.
+
+## IDEs
+
+![Editors](/img/developer-survey-2020/editors.png)
```

Comment:
vastausta -> responses
itse tehty -> own editor?
Määrä -> could be removed as we have the total in the bar label